### ISSUE TITLE
`imports-as-dependencies` doesn't detect missing `@types` packages

### ISSUE THREAD

**Issue by DetachHead** [gajus/eslint-plugin-jsdoc#1107]
thanks for adding the rule i suggested in #896! i noticed that it doesn't seem to detect `@types/` packages though

## Expected behavior
error because `@types/eslint` is not in `dependencies` or `devDependencies`
## Actual behavior
no error

## ESLint Config

```js
module.exports = {
    // ...
    rules: {
        'jsdoc/imports-as-dependencies': 'error'
    }
}
```

### source code
```js
/** @type {import('eslint').Linter.ParserOptions} */
const foo = {}
```
### package.json
```json
{
  "dependencies": {
    "eslint": "^8.13.0"
  },
  "devDependencies": {}
}
```

## Environment

- Node version: 19.7.0
- ESLint version 
- `eslint-plugin-jsdoc` version: 46.2.0

**Comment by brettz9** [comment]
Ugh, yeah, I really hadn't thought this through.  I'm not sure how much detection we want to be doing, e.g., to know whether to check for `types` in `package.json`, not to mention `typeRoots` or whatever.

**Comment by brettz9** [comment]
I'm thinking I may need to remove the rule unless TypeScript provides an API for detection, since there is quite a bit involved here--e.g., `types` within `exports`, `typeRoots`, etc.

**Comment by DetachHead** [comment]
[import/no-extraneous-dependencies](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md) is able to detect this, maybe its worth looking at the source code for that rule to see how it works

**Comment by brettz9** [comment]
That appears to be able to look for type imports, but not checking whether declaration files are present.

**Comment by brettz9** [comment]
Ok, I've submitted #1108 which should handle this. It doesn't do caching as it probably should, nor does it take into account `typeRoots` or use of `types` within `exports`, but it should provide an easy fix for the current problem at least.

**Comment by DetachHead** [comment]
@brettz9 looks like it now errors on imports from packages that come with type definitions (ie. does not need a `@types` package)
```js
/** @type {import('typescript').Program} */ // error  import points to package which is not found in dependencies  jsdoc/imports-as-dependencies
const foo = {}
```

in this case, i have `typescript` in `devDependencies` but there's no need for `@types/typescript` to be installed

**Comment by brettz9** [comment]
With the exception of TypeScript itself (for which I can supply a fix), it should still be working for packages with their own types. TypeScript just doesn't have `types` in `package.json` as other packages should.

**Comment by DetachHead** [comment]
looks like typescript uses `typings` instead of `types`, which is apparently an alias for the same thing, see https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

> Note that the `"typings"` field is synonymous with `types`, and could be used as well.
